proxyclient/m1n1/hv/__init__.py

```diff
@@ -1564,8 +1522,10 @@ def load_hook(data, segname, size, fileoff, dest):
 
         print(f"Physical memory: 0x{phys_base:x} .. 0x{mem_top:x}")
         print(f"Guest region start: 0x{guest_base:x}")
-
-        self.entry = macho.entry - macho.vmin + guest_base
+        if(is_macho_bin == True):
+            self.entry = macho_data.entry - macho_data.vmin + guest_base
```

Comment:
Why not just pass an `entry` parameter instead of the macho_data thing? That would be relative to the guest_base. Then here you just need to add guest_base, and you don't need any specific logic for macho/not.

At that point, this entire function can just become `def load_raw(self, data, entry=0x800):` itself, instead of `load_common`. Then `load_macho` can just call `load_raw` directly.

Reply:
resolved with the refactor